src/main/java/org/jabref/gui/texparser/CitationsDisplay.java

```diff
@@ -52,4 +55,41 @@ public Node getDisplayGraphic(Citation item) {
 
         return new VBox(contextBox, dataBox);
     }
+
+    private Tooltip getDisplayTooltip(Citation item) {
+        TextFlow tooltipText = new TextFlow();
+        String lineText = item.getLineText();
+        tooltipText.getChildren().setAll(new Text(lineText.substring(0, item.getColStart())));
+        tooltipText.getChildren().add(TooltipTextUtil.createText(lineText.substring(item.getColStart(), item.getColEnd()), TooltipTextUtil.TextType.BOLD));
+        tooltipText.getChildren().add(new Text(lineText.substring(item.getColEnd())));
+
+        Tooltip tooltip = new Tooltip();
+        tooltip.setContentDisplay(ContentDisplay.GRAPHIC_ONLY);
+        tooltip.setGraphic(tooltipText);
+        tooltip.setMaxHeight(10);
+        tooltip.maxWidthProperty().bind(this.widthProperty().subtract(85));
+        tooltip.setWrapText(true);
+
+        return tooltip;
+    }
+
+    class ListCitationsCellFactory implements Callback<ListView<Citation>, ListCell<Citation>> {
```

Comment:
Is there a reason why you switched to your own implementation of the CellFactory ?

Reply:
Because tooltips in `ViewModelListCellFactory` get a String as a parameter...

https://github.com/JabRef/jabref/blob/97f6e679de0511ce9b20e363d91400608e2b6aea/src/main/java/org/jabref/gui/util/ViewModelListCellFactory.java#L77-L80

https://github.com/JabRef/jabref/blob/97f6e679de0511ce9b20e363d91400608e2b6aea/src/main/java/org/jabref/gui/util/ViewModelListCellFactory.java#L165-L170

... and I think that we need a slightly more complex implementation:

https://github.com/JabRef/jabref/blob/97f6e679de0511ce9b20e363d91400608e2b6aea/src/main/java/org/jabref/gui/texparser/CitationsDisplay.java#L59-L74

Reply:
What about extending the factory e.g. by creating an overload for the ToolTip Object as parameter?

Reply:
Good point. I will work on it!